File: minimum-element-in-bst/mini_element_in_bst.py
#User function Template for python3

##Structure of the node

class Node:
    data=0
    left=None
    right=None


def minValue(root):
    if root is None:
        return -1
    elif (root.left is None and root.right is None):
        return root.data
    elif (root.left is None):
        return root.data
    else:
        return minValue(root.left)

File: minimum-element-in-bst/miniElementInBST.java
/* A binary search tree node class

class Node
{
	int data;
	Node left,right;

	Node(int d)
	{
		data = d;
		left = right = null;
	}
}
*/
// your task is to complete the below function
class Tree {
  int minValue(Node node) {
    if (node == null) {
      return -1;
    } else if (node.left == null && node.right == null) {
      return node.data;
    } else if (node.left == null) {
      return node.data;
    } else {
      return minValue(node.left);
    }
  }
}
